ecosystem/indexer-grpc/indexer-grpc-fullnode/src/stream_coordinator.rs

```diff
@@ -72,22 +77,25 @@ impl IndexerStreamCoordinator {
     /// 2. Convert transactions to rust objects (for example stringifying move structs into json)
     /// 3. Convert into protobuf objects
     /// 4. Encode protobuf objects (base64)
-    pub async fn process_next_batch(&mut self) -> Vec<Result<EndVersion, Status>> {
+    pub async fn process_next_batch(&mut self, pool: Arc<Pool>) -> Vec<Result<EndVersion, Status>> {
         let ledger_chain_id = self.context.chain_id().id();
         let mut tasks = vec![];
         let batches = self.get_batches().await;
         let output_batch_size = self.output_batch_size;
 
+        let pool = pool.clone();
         for batch in batches {
             let context = self.context.clone();
+            let pool = pool.clone();
             let ledger_version = self.highest_known_version;
             let transaction_sender = self.transactions_sender.clone();
-
             let task = tokio::spawn(async move {
                 // Fetch and convert transactions from API
                 let raw_txns =
                     Self::fetch_raw_txns_with_retries(context.clone(), ledger_version, batch).await;
-                let api_txns = Self::convert_to_api_txns(context, raw_txns).await;
+                let pool = pool.clone();
+                let api_txns = Self::convert_to_api_txns(context, raw_txns, pool.clone()).await;
```

Comment:
I see the batch size is 500 at most https://github.com/aptos-labs/aptos-core/blob/0e3956259aa7c823e27e16f379d9e63233e6ed6f/crates/indexer/src/indexer/fetcher.rs#L16

Can you try enlarging that to 10k and see how much it's faster?